x/evm/module.go

```diff
@@ -18,71 +22,87 @@ import (
 	"github.com/cosmos/ethermint/x/evm/types"
 )
 
-var _ module.AppModuleBasic = AppModuleBasic{}
-var _ module.AppModule = AppModule{}
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
 
-// AppModuleBasic struct
+// AppModuleBasic defines the basic application module used by the evm module.
 type AppModuleBasic struct{}
 
-// Name for app module basic
+// Name returns the evm module's name.
 func (AppModuleBasic) Name() string {
 	return types.ModuleName
 }
 
-// RegisterCodec registers types for module
-func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
-	types.RegisterCodec(cdc)
+// RegisterLegacyAminoCodec performs a no-op as the evm module doesn't support amino.
+func (AppModuleBasic) RegisterLegacyAminoCodec(_ *codec.LegacyAmino) {
 }
 
-// DefaultGenesis is json default structure
-func (AppModuleBasic) DefaultGenesis() json.RawMessage {
-	return types.ModuleCdc.MustMarshalJSON(types.DefaultGenesisState())
+// DefaultGenesis returns default genesis state as raw bytes for the evm
+// module.
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONMarshaler) json.RawMessage {
+	return cdc.MustMarshalJSON(types.DefaultGenesisState())
 }
 
 // ValidateGenesis is the validation check of the Genesis
-func (AppModuleBasic) ValidateGenesis(bz json.RawMessage) error {
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, _ client.TxEncodingConfig, bz json.RawMessage) error {
 	var genesisState types.GenesisState
-	err := types.ModuleCdc.UnmarshalJSON(bz, &genesisState)
-	if err != nil {
-		return err
+	if err := cdc.UnmarshalJSON(bz, &genesisState); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
 	}
 
 	return genesisState.Validate()
 }
 
-// RegisterRESTRoutes Registers rest routes
-func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
+// RegisterRESTRoutes performs a no-op as the EVM module doesn't expose REST
+// endpoints
+func (AppModuleBasic) RegisterRESTRoutes(_ client.Context, _ *mux.Router) {
 }
 
-// GetQueryCmd Gets the root query command of this module
-func (AppModuleBasic) GetQueryCmd(cdc *codec.Codec) *cobra.Command {
-	return cli.GetQueryCmd(types.ModuleName, cdc)
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the evm module.
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+	if err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)); err != nil {
+		panic(err)
+	}
 }
 
-// GetTxCmd Gets the root tx command of this module
-func (AppModuleBasic) GetTxCmd(cdc *codec.Codec) *cobra.Command {
+// GetTxCmd returns nil as the evm module doesn't support transactions through the CLI.
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
```

Comment:
this one returns nil but no longer supported, is it safe to remove? otherwise it might be called and cause nil pointer error

Reply:
it needs to fulfil the module interface so it is ok to return nil